packages/gatsby/src/schema/schema.js

```diff
@@ -117,9 +117,21 @@ const processTypeComposer = async ({
 
 const addTypes = ({ schemaComposer, types, parentSpan }) => {
   types.forEach(typeOrTypeDef => {
-    console.log(typeOrTypeDef)
     if (typeof typeOrTypeDef === `string`) {
-      schemaComposer.addTypeDefs(typeOrTypeDef)
+      const addedTypes = schemaComposer.addTypeDefs(typeOrTypeDef)
+      addedTypes.forEach(type => {
+        const { GraphQLInterfaceType, GraphQLUnionType } = require(`graphql`)
```

Comment:
This might not be the best place to do this -- I feel we now have multiple places where we do some kind of initial processing of types? But we can take a second look at this later.
I added `addSchemaMustHaveType` because in the tests the custom abstract types from SDL would be stripped from the built schema otherwise.
And `setResolveType` is similarly needed -- but this means that we only allow custom abstract types that combine types that implement Node. Is that restriction ok? Otherwise we'd have to provide some other way to add a type resolver, but I'm not sureI'd want that.